registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/validator/BiometricsSignatureValidatorTest.java

```diff
@@ -0,0 +1,171 @@
+		jwtSignatureVerifyResponseDto1.setSignatureValid(false);
+		jwtSignatureVerifyResponseDto1.setTrustValid("TRUST_CERT_PATH_VALID");
+
+		Mockito.when(env.getProperty(anyString())).thenReturn("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+		Map<String, String> metamap = new HashMap<>();
+		org.json.JSONArray jsonArray = new org.json.JSONArray();
+		org.json.JSONObject jsonObject1 = new org.json.JSONObject();
+		jsonObject1.put("Registration Client Version Number", "1.2.0");
+		jsonArray.put(0, jsonObject1);
+		metamap.put(JsonConstant.METADATA, jsonArray.toString());
+		Mockito.when(packetManagerService.getMetaInfo(anyString(), any(), any())).thenReturn(metamap);
+		Mockito.when(mapper.readValue(anyString(), any(Class.class)))
+				.thenReturn(new FieldValue("Registration Client Version Number", "1.2.0"));
+	}
+
+	@Test
+	public void signatureValidationTest()
+			throws JsonProcessingException, IOException, BaseCheckedException, JSONException {
+		ResponseWrapper1.setResponse(jwtSignatureVerifyResponseDto);
+		
+		PowerMockito.mockStatic(BiometricsSignatureHelper.class);
+		PowerMockito.when(BiometricsSignatureHelper.extractJWTToken(any())).thenReturn(constructedJwtToken);
+		
+		Mockito.when(registrationProcessorRestService.postApi(any(), any(), anyString(), any(), any()))
+				.thenReturn(ResponseWrapper1);
+		Mockito.when(mapper.writeValueAsString(any())).thenReturn("TkJna3Foa2lHOXcwQkFRc0ZBR");
+		Mockito.when(mapper.readValue(anyString(), eq(JWTSignatureVerifyResponseDto.class)))
+				.thenReturn(jwtSignatureVerifyResponseDto);
+		biometricsSignatureValidator.validateSignature(id, process, biometricRecord);
+	}
+
+	@Test(expected = BiometricSignatureValidationException.class)
+	public void signatureValidationFailedTest()
+			throws JsonProcessingException, IOException, BaseCheckedException, JSONException {
+		ResponseWrapper1.setResponse(jwtSignatureVerifyResponseDto1);
+		
+		PowerMockito.mockStatic(BiometricsSignatureHelper.class);
+		PowerMockito.when(BiometricsSignatureHelper.extractJWTToken(any())).thenReturn(constructedJwtToken);
+		
+		Mockito.when(registrationProcessorRestService.postApi(any(), any(), anyString(), any(), any()))
+				.thenReturn(ResponseWrapper1);
+		Mockito.when(mapper.writeValueAsString(any())).thenReturn("TkJna3Foa2lHOXcwQkFRc0ZBR");
+		Mockito.when(mapper.readValue(anyString(), eq(JWTSignatureVerifyResponseDto.class)))
+				.thenReturn(jwtSignatureVerifyResponseDto1);
+		biometricsSignatureValidator.validateSignature(id, process, biometricRecord);
+	}
+
+	@Test(expected = BiometricSignatureValidationException.class)
+	public void headerCertificateValidationFailedTest()
+			throws JsonProcessingException, IOException, BaseCheckedException, JSONException {
+		PowerMockito.mockStatic(BiometricsSignatureHelper.class);
+		PowerMockito.when(BiometricsSignatureHelper.extractJWTToken(any()))
+				.thenThrow(BiometricSignatureValidationException.class);
+		biometricsSignatureValidator.validateSignature(id, process, biometricRecord1);
```

Comment:
Have 2 more unit test cases
1. Backward compatibility of reg client version
2. Keep all exception segments and return invalid signature from jwt verfication api and see if no exception is thrown

Reply:
added these 2 test cases.